pkg/registry/apis/dashboard/authorizer.go

```diff
@@ -42,38 +44,84 @@ func GetAuthorizer(ac accesscontrol.AccessControl, l log.Logger) authorizer.Auth
 			}
 
-			switch attr.GetVerb() {
-			case "list", "search":
-				// Detailed read permissions are handled by authz, this just checks whether the user can ready *any* dashboard
-				ok, err := ac.Evaluate(ctx, user, accesscontrol.EvalPermission(dashboards.ActionDashboardsRead))
-				if !ok || err != nil {
-					return authorizer.DecisionDeny, "can not read any dashboards", err
-				}
-			case "create":
-				// Detailed create permissions are handled by authz, this just checks whether the user can create *any* dashboard
-				ok, err := ac.Evaluate(ctx, user, accesscontrol.EvalPermission(dashboards.ActionDashboardsCreate))
-				if !ok || err != nil {
-					return authorizer.DecisionDeny, "can not create any dashboards", err
-				}
-			case "get":
-				ok, err := ac.Evaluate(ctx, user, accesscontrol.EvalPermission(dashboards.ActionDashboardsRead, dashboards.ScopeDashboardsProvider.GetResourceScopeUID(attr.GetName())))
-				if !ok || err != nil {
-					return authorizer.DecisionDeny, "can not view dashboard", err
-				}
-			case "update", "patch":
-				ok, err := ac.Evaluate(ctx, user, accesscontrol.EvalPermission(dashboards.ActionDashboardsWrite, dashboards.ScopeDashboardsProvider.GetResourceScopeUID(attr.GetName())))
-				if !ok || err != nil {
-					return authorizer.DecisionDeny, "can not edit dashboard", err
-				}
-			case "delete":
-				ok, err := ac.Evaluate(ctx, user, accesscontrol.EvalPermission(dashboards.ActionDashboardsDelete, dashboards.ScopeDashboardsProvider.GetResourceScopeUID(attr.GetName())))
-				if !ok || err != nil {
-					return authorizer.DecisionDeny, "can not delete dashboard", err
-				}
-			default:
-				l.Info("unknown verb", "verb", attr.GetVerb())
-				return authorizer.DecisionDeny, "unsupported verb", nil // Unknown verb
+			// Determine if this is a library panel or dashboard resource
+			resource := attr.GetResource()
+			isLibraryPanel := resource == dashv0.LIBRARY_PANEL_RESOURCE
+
+			if isLibraryPanel {
+				return authorizeLibraryPanel(ctx, ac, user, attr)
+			} else {
+				return authorizeDashboard(ctx, ac, user, attr)
 			}
-			return authorizer.DecisionAllow, "", nil
 		})
 }
+
+func authorizeLibraryPanel(ctx context.Context, ac accesscontrol.AccessControl, user identity.Requester, attr authorizer.Attributes) (authorizer.Decision, string, error) {
+	switch attr.GetVerb() {
+	case "list", "search":
+		// Detailed read permissions are handled by authz, this just checks whether the user can ready *any* library panel
+		ok, err := ac.Evaluate(ctx, user, accesscontrol.EvalPermission(libraryelements.ActionLibraryPanelsRead))
+		if !ok || err != nil {
+			return authorizer.DecisionDeny, "can not read any library panels", err
+		}
+	case "create":
+		// Detailed create permissions are handled by authz, this just checks whether the user can create *any* library panel
+		ok, err := ac.Evaluate(ctx, user, accesscontrol.EvalPermission(libraryelements.ActionLibraryPanelsCreate))
```

Comment:
Where is this extra check carried out?

Reply:
Create isn't supported yet via /apis, will need to follow-up with another pr to add support

Reply:
Ok, makes sense. I think it would be safer to return access denied for create for now and then add this evaluator once we also add a check with scope. Just so that we don't forget to add the other access check. Or would that not work?